doc/static/css/zephyr.css

```diff
@@ -1,56 +1,78 @@
-/* -- Extra CSS styles for Zephyr content (RTD theme) ----------------------- */
-
-/*  make .. hlist:: tables fill the page */
-table.hlist {
-    width: 95% !important;
+:root {
+  --docset-color: #a67dfb;
 }
 
-/*  override rtd theme white-space no-wrap in table heading and content  */
-th,td {
-    white-space: normal !important;
+/* home page grid display */
+.grid {
```

Comment:
We can't inherit Zephyr customizations blindly, as we use our template. This grid CSS is just for the home page grid (which by the way should not be into the generic CSS file, but we'd need to fix that in Zephyr first).

Reply:
This means though that formatting will be broken every time Zephyr does a new customization.
With including Zephyr's custom stylesheet, it might not fit our style, but at least it might work out-of-the-box.

Reply:
Zephyr may eventually switch to another theme, so customizations may not be relevant to us because NCS uses its own theme (with its own set of customizations). We only need to take care of Zephyr content, not its theme as we render Zephyr content with our custom theme. Any CSS customizations to the theme need to be on our side. The grid CSS is a particular case, it's only used by the home page.

Reply:
Sure, if Zephyr switches to another theme, some customizations might not be usable anymore. Others will still be, though.
And at the moment, we're losing formatting that used to work just fine.

For example, keys (kbd):
Now: http://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-4160/zephyr/application/index.html#running-in-an-emulator
Before: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/application/index.html#running-in-an-emulator

Or rst-columns:
Now: http://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-4160/zephyr/guides/coccinelle.html#getting-coccinelle
Before: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/zephyr/guides/coccinelle.html#getting-coccinelle

Both of these are simple css customizations that would probably work even in a different theme.
If we decide to not include them automatically, someone needs to constantly monitor the customization file and carry changes over in each upmerge.

Reply:
The customizations we want to keep need to go to `sphinx_ncs_theme` so that all docsets are consistent.

Reply:
The ones that we use in other doc sets, yes. Not the ones that Zephyr introduces and that we don't use anywhere in our docs (or don't even know about).

Reply:
Zephyr does some changes that will lead to inconsistencies between docsets. For example, code blocks (current docs):

![image](https://user-images.githubusercontent.com/25011557/112205827-834d1180-8c15-11eb-9f81-51c2dae581d3.png)
_nRF (green bg)_

![image](https://user-images.githubusercontent.com/25011557/112205862-8e07a680-8c15-11eb-975c-75d85f9acc27.png)
_Zephyr (gray bg)_

We are lucky that Zephyr and NCS use the same base template (sphinx_rtd_theme), otherwise importing its CSS would lead to unpredictable results. Content needs to be rendered consistently, this is why `sphinx_ncs_theme` has been created. Changing things in a single place results in a uniform change on all docsets.

Reply:
I don't see an issue in slight differences in rendering for Zephyr, since Zephyr IS different.
But in the end, it's up to you - if you will monitor all changes needed to get the Zephyr content to show up correctly, that is fine. I know it hasn't been done up to now. And since we cannot update the theme for docs that are already published, it's better to have basic support for new features, even if it doesn't fit our style 100%, than no support at all.